base/locks-mt.jl

```diff
@@ -12,33 +12,62 @@ export SpinLock
 ##########################################
 
 """
-    SpinLock()
+    abstract type AbstractSpinLock <: AbstractLock end
 
-Create a non-reentrant, test-and-test-and-set spin lock.
+A non-reentrant, test-and-test-and-set spin lock.
 Recursive use will result in a deadlock.
 This kind of lock should only be used around code that takes little time
 to execute and does not block (e.g. perform I/O).
 In general, [`ReentrantLock`](@ref) should be used instead.
 
 Each [`lock`](@ref) must be matched with an [`unlock`](@ref).
-If [`!islocked(lck::SpinLock)`](@ref islocked) holds, [`trylock(lck)`](@ref trylock)
+If [`!islocked(lck::AbstractSpinLock)`](@ref islocked) holds, [`trylock(lck)`](@ref trylock)
 succeeds unless there are other tasks attempting to hold the lock "at the same time."
 
 Test-and-test-and-set spin locks are quickest up to about 30ish
 contending threads. If you have more contention than that, different
 synchronization approaches should be considered.
 """
-mutable struct SpinLock <: AbstractLock
+abstract type AbstractSpinLock <: AbstractLock end
+
+"""
+    SpinLock() <: AbstractSpinLock
+
+Spinlocks are not padded, and so may suffer from false sharing.
```

Comment:
since the documentation for `AbstractSpinLock` contains a lot of useful information, i think we should refer users there, and also cross-reference the new `PaddedSpinLock`. Perhaps we could also add a sentence saying when/why one might prefer `SpinLock` to `PaddedSpinLock` (and vice versa)
```suggestion
Spinlocks are not padded, and so may suffer from false sharing.
See also [`PaddedSpinLock`](@ref)

See the documentation for [`AbstractSpinLock`](@ref) regarding correct usage.
```